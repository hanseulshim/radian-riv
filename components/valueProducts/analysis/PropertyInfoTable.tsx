import React, { useState, useCallback } from 'react'
import { useValueProduct } from 'context/ValueProductProvider'
import Table from 'components/common/Table'

interface Props {
  views: string[]
  currentView: string
  setCurrentView: React.Dispatch<React.SetStateAction<string>>
}

export default function PropertyInfoTable({
  views,
  currentView,
  setCurrentView
}: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])
  const { propertyInfo } = useValueProduct()

  const fetchData = useCallback(async () => {
    const tableData = [
      {
        ...propertyInfo
      }
    ]
    setColumns([
      {
        Header: 'Pool Name',
        accessor: 'poolName'
      },
      {
        Header: 'Loan Number',
        accessor: 'loanNumber'
      },
      {
        Header: 'ID',
        accessor: 'id'
      },
      {
        Header: 'Subject',
        accessor: row => {
          return `${row['address']}, ${row['city']}, ${row['state']}`
        }
      },
      {
        Header: 'Zip',
        accessor: 'zip'
      },
      {
        Header: 'Radian Interactive Value Date',
        accessor: 'rivDate'
      },
      {
        Header: 'Calculated Price',
        accessor: 'calculatedPrice'
      },
      {
        Header: 'As of Date',
        accessor: 'asOfDate'
      }
    ])
    setData(tableData)
  }, [propertyInfo])

  return (
    <div className="property-tabs">
      <Table columns={columns} data={data} fetchData={fetchData} />
      <ul className="tab-container">
        {views.map(view => (
          <li
            key={view}
            onClick={() => setCurrentView(view)}
            className={`${view === currentView ? 'active ' : ''}`}
          >
            {view}
          </li>
        ))}
      </ul>
    </div>
  )
}
