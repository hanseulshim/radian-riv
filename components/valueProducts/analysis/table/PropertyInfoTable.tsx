import React, { useState, useCallback } from 'react'
import { useOrder } from 'context/OrderProvider'
import Table from 'components/common/Table'
import { formatPrice } from 'utils'

interface Props {
  views: string[]
  currentView: string
  setCurrentView: React.Dispatch<React.SetStateAction<string>>
}

export default function PropertyInfoTable({
  views,
  currentView,
  setCurrentView
}: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])
  const { order } = useOrder()

  const fetchData = useCallback(async () => {
    const tableData = [
      {
        ...order
      }
    ]
    setColumns([
      {
        Header: 'Pool Name',
        accessor: 'poolName',
        width: 120
      },
      {
        Header: 'Loan Number',
        accessor: 'loanNumber',
        width: 125
      },
      {
        Header: 'ID',
        accessor: 'id',
        width: 125
      },
      {
        Header: 'Subject',
        accessor: row => {
          return `${row['address']}, ${row['city']}, ${row['state']}`
        },
        width: 150
      },
      {
        Header: 'Zip',
        accessor: 'zip',
        width: 95
      },
      {
        Header: 'Radian Interactive Value Date',
        accessor: 'rivDate',
        width: 180
      },
      {
        Header: 'Calculated Price',
        accessor: row => {
          return formatPrice(row.calculatedPrice)
        },
        width: 100
      },
      {
        Header: 'As of Date',
        accessor: 'asOfDate',
        width: 125
      }
    ])
    setData(tableData)
  }, [order])

  return (
    <div className="property-tabs">
      <Table columns={columns} data={data} fetchData={fetchData} />
      <ul className="tab-container">
        {views.map(view => (
          <li
            key={view}
            onClick={() => setCurrentView(view)}
            className={`${view === currentView ? 'active ' : ''}`}
          >
            {view}
          </li>
        ))}
      </ul>
    </div>
  )
}
