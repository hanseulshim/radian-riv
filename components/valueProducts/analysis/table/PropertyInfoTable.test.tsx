import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import PropertyInfoTable from './PropertyInfoTable'

describe('PropertyInfoTable', () => {
  let component
  beforeAll(() => {
    const props = {
      views: ['one', 'two', 'three'],
      currentView: 'one',
      setCurrentView: () => {}
    }
    component = shallow(<PropertyInfoTable {...props} />)
  })
  it('get table container', () => {
    expect(component.find('.property-tabs')).toHaveLength(1)
  })
  it('get tab container', () => {
    expect(component.find('.tab-container')).toHaveLength(1)
  })
  it('have three total tags and one active tab', () => {
    expect(component.find('.tab-container > .active')).toHaveLength(1)
    expect(component.find('.tab-container > li')).toHaveLength(3)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
