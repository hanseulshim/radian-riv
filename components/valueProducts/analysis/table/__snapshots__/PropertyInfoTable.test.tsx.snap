// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyInfoTable should match snapshot 1`] = `
<div
  className="property-tabs"
>
  <Table
    columns={Array []}
    data={Array []}
    fetchData={[Function]}
  />
  <ul
    className="tab-container"
  >
    <li
      className="active "
      key="one"
      onClick={[Function]}
    >
      one
    </li>
    <li
      className=""
      key="two"
      onClick={[Function]}
    >
      two
    </li>
    <li
      className=""
      key="three"
      onClick={[Function]}
    >
      three
    </li>
  </ul>
</div>
`;
